src/main/java/com/exasol/sql/expression/rendering/BooleanExpressionRenderer.java

```diff
@@ -90,34 +92,16 @@ protected void appendOperand(final ValueExpression leftOperand) {
 
     @Override
     public void leave(final Comparison comparison) {
-        if (!comparison.isRoot()) {
-            endParenthesis();
-        }
-    }
-
-    @Override
-    public void visit(final Like like) {
-        connect(like);
-        if (!like.isRoot()) {
-            startParenthesis();
+        if (comparison instanceof LikeComparison) {
```

Comment:
This doesn't look right. Why is this logic in the leave part?  :D 

And I see that you created a new interface `ComparisonVisitor`. Why is the rendering still here?

Reply:
I moved it above.

Reply:
I added the ComparisonVisitor for external use. For the rendering, at the moment, this solution is more readable and muich shorter. In the future, it might be better to use a visitor, but now it would only add complexity. (it's only one if right now...)

Reply:
ok. I changed it.